### PR TITLE
Ensure secondary has sufficient available pods before switching over in blue-green deployment

### PR DESCRIPTION
**Pull request by simoncochrane**

### CODE REVIEW

File: deploy/bluegreen.go [hunk 1]
```diff
@@ -271,63 +276,89 @@ func (bgd *BlueGreenDeployer) ensureHasPods(target TemplateRolloutTarget, next s
...
-		return state.Single(bgd.waitForAllPods(secondary, next))
+		ok, err := bgd.checkPods(secondary, primaryNum)
+		if err != nil {
+			return state.Error(errors.Wrapf(err, "failed to check pods while scaling up sceondary %s", secondary.Name()))
+		}
+
+		if ok {
+			return state.Single(next)
+		}
+		return state.After(15*time.Second, bgd.scaleUpSecondary(primary, secondary, next))
```
Comment (jh-sz) [inline deploy/bluegreen.go]: Trying to trace through this code, but don't fully understand it, is the 15 seconds the polling time or expiry time?

Comment (simoncochrane) [reply deploy/bluegreen.go]: It's like a polling time. It is saying to perform the `scaleUpSecondary` step in 15 seconds.

Comment (jh-sz) [reply deploy/bluegreen.go]: so where's the expiry defined? I am mainly referring to this comment here
https://github.com/nearmap/kcd/pull/45/files#diff-b6934004e468df37ebb5fe2caa7dd059R276

Comment (simoncochrane) [reply deploy/bluegreen.go]: That comment is no longer correct. A context with deadline is now set when the workflow begins, which causes the rollout to fail if reached.

**Pull request merged**